cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -596,10 +585,78 @@ instance ToJSON DelegationsAndRewards where
 
 -- Helpers
 
+calcEraInMode
+  :: CardanoEra era
+  -> ConsensusMode mode
+  -> ExceptT ShelleyQueryCmdError IO (EraInMode era mode)
+calcEraInMode era mode=
+  hoistMaybe (ShelleyQueryCmdEraConsensusModeMismatch (anyCardanoEra era) (AnyConsensusMode mode))
+                   $ toEraInMode era mode
+
+determineEra
+  :: ConsensusModeParams mode
+  -> LocalNodeConnectInfo mode
+  -> ChainPoint
+  -> ExceptT ShelleyQueryCmdError IO AnyCardanoEra
+determineEra cModeParams localNodeConnInfo point =
+  case consensusModeOnly cModeParams of
+    ByronMode -> return $ AnyCardanoEra ByronEra
+    ShelleyMode -> return $ AnyCardanoEra ShelleyEra
+    CardanoMode -> do
+      eraQ <- liftIO . queryNodeLocalState localNodeConnInfo (Just point)
```

Comment:
We can pass `Nothing` for the `point` here and simplify things a bit, both here but also for the callers which will not need to query for the tip at all.

Reply:
We can put `Nothing` but then we would be querying the tip twice (could this potentially also return different tips?) as we need a `ChainTip` in `executeQuery`.